digdag-standards/src/main/java/io/digdag/standards/operator/MailOperatorFactory.java

```diff
@@ -204,19 +169,81 @@ public PasswordAuthentication getPasswordAuthentication() {
 
                     msg.setContent(multipart);
                 }
+
                 Transport.send(msg);
             }
             catch (MessagingException | IOException ex) {
-                throw new RuntimeException(ex);
+                throw new TaskExecutionException(ex, TaskExecutionException.buildExceptionErrorConfig(ex));
             }
 
             return TaskResult.empty(request);
         }
 
-        private String getFlatOrNested(Config config, String key)
+        private List<AttachConfig> attachConfigs(Config params)
         {
-            return config.getNestedOrGetEmpty("mail").getOptional(key, String.class)
-                .or(() -> config.get(key, String.class));
+            return params.getListOrEmpty("attach_files", Config.class)
+                    .stream()
+                    .map((a) -> {
+                        String path = a.get("path", String.class);
+                        return ImmutableAttachConfig.builder()
+                                .path(path)
+                                .fileName(
+                                        a.getOptional("filename", String.class)
+                                                .or(path.substring(Math.max(path.lastIndexOf('/'), 0)))
+                                )
+                                .contentType(
+                                        a.getOptional("content_type", String.class)
+                                                .or("application/octet-stream")
+                                )
+                                .build();
+                    })
+                    .collect(Collectors.toList());
+        }
+
+        private Session createSession(SecretProvider secrets, Config params)
+        {
+            // Use only _either_ user supplied smtp configuration _or_ system smtp configuration to avoid leaking credentials
+            // by e.g. connecting to a user controlled host and handing over user/password in base64 plaintext.
+            SmtpConfig smtpConfig = userSmtpConfig(secrets, params)
```

Comment:
No, we do _not_ want to merge configuration values from the system config if the user has specified a host.